src/tests/AabbTest.cpp

```diff
@@ -0,0 +1,112 @@
+                                      const Mn::Matrix4& xform) {
+  std::vector<Mn::Vector3> corners;
+  corners.push_back(xform.transformPoint(range.frontBottomLeft()));
+  corners.push_back(xform.transformPoint(range.frontBottomRight()));
+  corners.push_back(xform.transformPoint(range.frontTopLeft()));
+  corners.push_back(xform.transformPoint(range.frontTopRight()));
+
+  corners.push_back(xform.transformPoint(range.backTopLeft()));
+  corners.push_back(xform.transformPoint(range.backTopRight()));
+  corners.push_back(xform.transformPoint(range.backBottomLeft()));
+  corners.push_back(xform.transformPoint(range.backBottomRight()));
+
+  Mn::Range3D transformedBB{Mn::Math::minmax<Mn::Vector3>(corners)};
+
+  return transformedBB;
+}
+
+struct GeoTest : Cr::TestSuite::Tester {
+  explicit GeoTest();
+  // tests
+  void aabb();
+  // benchmarks
+  void getTransformedBB_standard();
+  void getTransformedBB();
+
+  std::vector<Mn::Matrix4> xforms_;
+  // number of transformations
+  const unsigned int numCases_ = 50000;
+  // the batch size when running benchmarks
+  const unsigned int iterations_ = 10;
+  Mn::Range3D box_ = Mn::Range3D{Mn::Vector3{-10.0, -10.0, -10.0},
+                                 Mn::Vector3{10.0, 10.0, 10.0}};
+};
+
+GeoTest::GeoTest() {
+  // clang-format off
+  addTests({&GeoTest::aabb});
+  addBenchmarks({&GeoTest::getTransformedBB_standard,
+                 &GeoTest::getTransformedBB}, 10);
+  // clang-format on
+
+  // Generate N transformations (random positions and orientations)
+  xforms_.reserve(numCases_);
+  for (unsigned int iTransform = 0; iTransform < numCases_; ++iTransform) {
+    // generate random translation
+    const Mn::Vector3 translation{
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f,
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f,
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f};
+
+    // assemble the transformation matrix, and push it back
+    xforms_.emplace_back(
+        Mn::Matrix4::from(esp::core::randomRotation().toMatrix(), translation));
+  }
+}
+
+void GeoTest::getTransformedBB_standard() {
+  Mn::Range3D aabb;
+  CORRADE_BENCHMARK(iterations_) for (auto& xform : xforms_) {
+    aabb = Test::getTransformedBB_standard(box_, xform);
+  }
+}
+
+void GeoTest::getTransformedBB() {
+  Mn::Range3D aabb;
+  CORRADE_BENCHMARK(iterations_) for (auto& xform : xforms_) {
+    aabb = esp::geo::getTransformedBB(box_, xform);
+  }
+}
+
+void GeoTest::aabb() {
+  // compute aabb for each box using standard method and library method
+  // respectively.
+  // compare the results, which should be identical
+  for (auto& xform : xforms_) {
+    Mn::Range3D aabbControl = Test::getTransformedBB_standard(box_, xform);
+    Mn::Range3D aabbTest = esp::geo::getTransformedBB(box_, xform);
+
+    float eps = 1e-8f;
+    CORRADE_COMPARE_WITH((aabbTest.min() - aabbControl.min()).dot(), 0.0,
```

Comment:
@mosra: This is what I did now with the `CORRADE_COMPARE_WITH(...)`. Finally it is the same as the 1st version I did in GTEST.

Reply:
Should work like this I think:

```cpp
CORRADE_COMPARE__WITH(aabbTest.min(), aabbControl.min(), 
    Cr::TestSuite::Compare::around(Mn::Vector3{1e-12f}));
CORRADE_COMPARE__WITH(aabbTest.max(), aabbControl.max(), 
    Cr::TestSuite::Compare::around(Mn::Vector3{1e-12f}));
```

Apologies that this took so many iterations, heh :)

Reply:
Sorry, it will not work.

Should be only one underscore before WITH.

Reply:
Yeah of course (edited). The main suggestion was about `around(Mn::Vector3{1e-12f})` instead of `around(1e-12f)`. Did *that* work?

Reply:
Just tested. Yes, it works now. Thank you.